esi_leap/api/controllers/v1/node.py

```diff
@@ -47,6 +56,16 @@ def get_all(self):
         nodes = ironic.get_node_list(context)
 
         node_collection = NodeCollection()
-        node_collection.nodes = [Node(name=n.name) for n in nodes]
+
+        for node in nodes:
+            offers = offer_obj.Offer.get_all({"resource_uuid": node.uuid},
+                                             context)
+            n = Node(name=node.name, uuid=node.uuid, owner=node.owner)
+            if offers:
+                n.offer_uuid = offers[0].uuid
```

Comment:
Showing one is reasonable, but I think it should either be:

* show a current offer, in which case you need to further update the offer filter to limit offers to a current time
* show all active and future offers

Either way, I think adding the start/end time would be useful.

Reply:
I decided to show one for now, but I added filters to make sure its only the currently available one. I can add a future offer uuids field if needed.